code/game/sim_manager/datums/simulator.dm

```diff
@@ -56,27 +51,15 @@
 
 /datum/simulator/proc/sim_turf_garbage_collection()
 
-	// initial grid needs an offset to the bottom left so it can get the most coverage within the users pov.
-	var/turf/sim_grid_start_pos = locate(sim_camera.x - 9,sim_camera.y - 9,1)
-	if(!sim_grid_start_pos)
+	var/area/sim_grid_area = get_area(sim_camera)
+	if(!sim_grid_area)
 		sim_reboot_state = FALSE
 		return
 
 	QDEL_LIST(delete_targets)
 
-	// 16x16 grid, clears from left to right like so
-	// the user's pov should be near the center of the grid.
-	/*
-	y:16| x: 1 2 3 4 ... 16
-	... | ...
-	y:2 | x: 1 2 3 4 ... 16
-	y:1 | x: 1 2 3 4 ... 16
-	*/
-	for (var/y_pos in 1 to GRID_CLEARING_SIZE)// outer y
-		for (var/x_pos in 1 to GRID_CLEARING_SIZE) // inner x
-			var/turf/current_grid = locate(sim_grid_start_pos.x + x_pos,sim_grid_start_pos.y + y_pos,sim_grid_start_pos.z)
-
-			current_grid.empty(/turf/open/floor/engine)
+	for(var/turf/open/floor/engine/engine_floor in sim_grid_area)
```

Comment:
this proc was intentionally not iterating through areas, because iterating through areas is actually an expensive `in world` check, which iterates through every turf in the world to check if it is in the given area

Reply:
huh.

Reply:
I tried to do a funny and used recursion.